lib/vector-buffers/src/topology/channel/sender.rs

```diff
@@ -13,6 +13,39 @@ use crate::{
     Bufferable, WhenFull,
 };
 
+/// Error related to sending an item via `BufferSender`.
+///
+/// This error type is meant to opaquely wrap the various error types returned by different buffer
+/// implementations in order to pass along the underlying error message to higher-level callers.
+#[derive(Debug)]
+pub struct SendError {
+    inner: Box<dyn error::Error + Send + Sync>,
```

Comment:
Could this re-use what is now `vector_core::Error` (moving it into `vector_common` probably)? The only difference is that `vector_core::Error` is `'static` too.

I would also suggest making this a newtype pattern since it is just a single field, but that's more bikeshedding than anything.

Reply:
It could definitely use a theoretical `vector_common::Error` internally, and be a newtype wrapper, totally.

Reply:
I ended up removing `SendError` entirely in ddd91a8d6 and switching purely to `crate::Result`, which is a crate-internal re-export of `vector_common::Result` after moving `vector_core::{Error, Result}` to `vector_common`.